src/hooks/async/marketplace.ts

```diff
@@ -0,0 +1,220 @@
+import { useActiveChainId } from "../../Provider";
+import { cacheKeys, createCacheKeyWithNetwork } from "../../utils/cache-keys";
+import { useQueryWithNetwork } from "../query-utils/useQueryWithNetwork";
+import { useAddress } from "../useAddress";
+import {
+  Marketplace,
+  MarketplaceFilter,
+  NewAuctionListing,
+  NewDirectListing,
+} from "@thirdweb-dev/sdk";
+import { useMutation, useQueryClient } from "react-query";
+import invariant from "tiny-invariant";
+
+/** **********************/
+/**     READ  HOOKS     **/
+/** **********************/
+
+/**
+ * Use this to get a list all listings from your marketplace contract.
+ *
+ * @example
+ * ```javascript
+ * const { data: listings, isLoading, error } = useMarketplaceListings(<YourMarketplaceContractInstance>, { start: 0, count: 100 });
+ * ```
+ *
+ * @param contract - an instace of a marketplace contract
+ * @param queryParams - query params to pass to the query for the sake of pagination
+ * @returns a response object that includes an array of listings
+ * @beta
+ */
+export function useMarketplaceListings(
```

Comment:
I wonder if we can just call this `useAllListings()` (and then export/import it like `@thirdweb-dev/react/marketplace` (we could make this export logic work work)

Reply:
I like that - it might be a bit confusing for NFT modules though - ill play around and see what it's like